code/src/components/TodoList.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { selectTodoList, completeAll, deleteAll } from 'reducers/todoSlice';
+import TodoItem from './TodoItem';
+
+const TodoList = () => {
+  const todoList = useSelector(selectTodoList)
+  const dispatch = useDispatch()
+  const allDone = () => {
+    dispatch(completeAll())
+  }
+  const allDeleted = () => {
+    dispatch(deleteAll())
+  }
+
+  return (
+    <>
+      <div>
+        <button type="button" onClick={allDone}>complete</button>
+        <button type="button" onClick={allDeleted}>delete</button>
```

Comment:
as you know these are currently not workin, which is too bad :). Not sure why not tho, but for me I had to use it in the component instead of importing it from the reducer. Like dispatch(todos.actions.deleteAll())